packages/@jsii/java-runtime/project/src/main/java/software/amazon/jsii/JsiiEngine.java

```diff
@@ -486,7 +486,11 @@ private Object invokeMethod(final Object obj, final Method method, final Object.
                 throw e;
             }
         } catch (InvocationTargetException e) {
-            throw new JsiiError(e.getTargetException());
+            if (e.getTargetException().getClass().equals(JsiiError.class)) {
+                throw new JsiiError(e.getTargetException());
```

Comment:
Could/should this be:

```suggestion
            if (e.getTargetException() instanceof JsiiError) {
                throw (JsiiError)(e.getTargetException();
```

I'm not sure there is much interest in re-wrapping that, it'll only bury the actual error location further down?